### PR TITLE
New feat : Upload image in torrent upload page & torrent detail

### PR DESCRIPTION
**Pull request by Klaiment**
With this update, here are the points that have been modified/added:

1. Fixed a Docker bug that prevented the correct mounting of the config.js file with Docker by explicitly indicating that it is a file and not a directory.

2. Added the image upload functionality requested in this report:
[link to the report](https://github.com/tdjsnelling/sqtracker/issues/14)

3. Also visible on the torrent detail page.

4. Added the French locales.

### CODE REVIEW

File: client/pages/torrent/[infoHash].js [hunk 1]
```diff
@@ -523,7 +523,10 @@ const Torrent = ({ token, torrent = {}, userId, userRole, uid, userStats }) => {
       userStats.ratio < Number(SQ_MINIMUM_RATIO)) ||
     (Number(SQ_MAXIMUM_HIT_N_RUNS) !== -1 &&
       userStats.hitnruns > Number(SQ_MAXIMUM_HIT_N_RUNS));
-
+  function isPngImage(data) {
```
Comment (tdjsnelling) [inline client/pages/torrent/[infoHash].js:526]: Please follow existing code style here and include blank lines before & after function

File: client/pages/torrent/[infoHash].js [hunk 2]
```diff
@@ -660,6 +664,20 @@ const Torrent = ({ token, torrent = {}, userId, userRole, uid, userStats }) => {
               : [getLocaleString("torrNo")],
         }}
       />
+      <Infobox mb={5}>
```
Comment (tdjsnelling) [inline client/pages/torrent/[infoHash].js:667]: This InfoBox should only be shown if `torrent.poster` exists, the same as the media info box

File: client/pages/torrent/[infoHash].js [hunk 3]
```diff
@@ -690,6 +708,7 @@ const Torrent = ({ token, torrent = {}, userId, userRole, uid, userStats }) => {
           </Box>
         </Infobox>
       )}
+      {console.log(torrent)}
```
Comment (tdjsnelling) [inline client/pages/torrent/[infoHash].js:711]: Please remove

File: client/pages/upload.js [hunk 4]
```diff
@@ -234,19 +235,59 @@ const Upload = ({ token, userId }) => {
...
+        reader.onload = async () => {
+          console.log(
+            `[DEBUG] Poster upload complete: ${reader.result.slice(0, 64)}...`
+          );
+          const [, posterB64] = reader.result.split("base64,");
+          setPosterFile({ name: file.name, b64: posterB64 });
+        };
+        reader.onerror = () => {
+          console.log(`[DEBUG] Poster upload error: ${reader.error}`);
+          // Gérer les erreurs, si nécessaire
```
Comment (tdjsnelling) [inline client/pages/upload.js:252]: Please remove any French comments

File: client/pages/upload.js [hunk 5]
```diff
@@ -381,6 +423,31 @@ const Upload = ({ token, userId }) => {
             </Text>
           )}
         </Box>
+        <Box mb={4}>
```
Comment (tdjsnelling) [inline client/pages/upload.js:426]: Please make this field optional. Currently I cannot submit the form if I do not upload an image. Please also move the field down to the bottom of the form, above tags

File: config.example.js [hunk 6]
```diff
@@ -74,13 +74,13 @@ module.exports = {
     SQ_EXTENSION_BLACKLIST: ["exe"],
 
     // The URL of your tracker site.
-    SQ_BASE_URL: "https://sqtracker.dev",
+    SQ_BASE_URL: "http://localhost",
```
Comment (tdjsnelling) [inline config.example.js:77]: Please revert changes to this file. This file is to serve as a production example, using localhost here is incorrect

File: docker-compose.dev.yml [hunk 7]
```diff
@@ -20,28 +21,35 @@ services:
     container_name: sq_mongodb
     image: mongo:6.0
     ports:
-      - "127.0.0.1:27017:27017"
+      - "0.0.0.0:27017:27017"
```
Comment (tdjsnelling) [inline docker-compose.dev.yml:24]: Please revert

File: client/pages/torrent/[infoHash].js [hunk 8]
```diff
@@ -660,6 +664,20 @@ const Torrent = ({ token, torrent = {}, userId, userRole, uid, userStats }) => {
               : [getLocaleString("torrNo")],
         }}
       />
+      <Infobox mb={5}>
+        <Text fontWeight={600} fontSize={1} textTransform="uppercase" mb={3}>
```
Comment (tdjsnelling) [inline client/pages/torrent/[infoHash].js:668]: Text style is incorrect here, please observe how `textTransform` is applied on the other headings

File: client/pages/upload.js [hunk 9]
```diff
@@ -234,19 +235,59 @@ const Upload = ({ token, userId }) => {
...
+      // Gérer les erreurs, si nécessaire
+    }
+  }, []);
 
   const { getRootProps, getInputProps, isDragActive } = useDropzone({
     onDrop,
     accept: { "application/x-bittorrent": [".torrent"] },
     maxFiles: 1,
   });
+  const {
```
Comment (tdjsnelling) [inline client/pages/upload.js:267]: Please add a sensible max. file size for this upload

File: client/pages/upload.js [hunk 10]
```diff
@@ -234,19 +235,59 @@ const Upload = ({ token, userId }) => {
...
+    return data.startsWith(pngHeader);
+  }
 
   const handleUpload = async (e) => {
     e.preventDefault();
     setLoading(true);
     const form = new FormData(e.target);
 
     try {
+      form.append("poster", posterFile.b64); // Ajoutez la valeur de l'image de l'affiche à la requête
```
Comment (tdjsnelling) [inline client/pages/upload.js:290]: Does this need to be appended to the form when you are submitting it below in the fetch request?

**Review by tdjsnelling — CHANGES_REQUESTED** [review]

File: api/src/controllers/torrent.js [hunk 11]
```diff
@@ -507,6 +508,7 @@ export const getTorrentsPage = async ({
         created: 1,
         freeleech: 1,
         tags: 1,
+        poster: 1,
```
[the remark this answers is not in the document]
Comment (Klaiment) [reply api/src/controllers/torrent.js:511]: The poster is retrieved on the torrent page so I need to retrieve it here

[the remark this answers is not in the document]
Comment (tdjsnelling) [reply api/src/controllers/torrent.js:511]: `getTorrentsPage` gets a page of torrents (25) to display in lists (search, browse etc.). Only `fetchTorrent` which fetches a single torrent to display on `/torrent/[infoHash]` needs to include the image. Sorry for confusing naming

[the remark this answers is not in the document]
Comment (Klaiment) [reply api/src/controllers/torrent.js]: fixed

**Comment by Klaiment** [comment]
All done.

**Comment by CodeMeZone** [comment]
Any ETA on this push?

**Comment by abiteman** [comment]
Also looking at an update on a potential push.

**Comment by tdjsnelling** [comment]
Sorry for the delay. Thanks for your changes, everything looks good now. Good job getting this implemented

**Pull request merged**